File: demo_visual/benchmark_matmul.py

import torch
import time

def main():
    size = 1024
    print(f"Benchmarking PyTorch MatMul [{size}, {size}] ...")
    
    a = torch.randn(size, size)
    b = torch.randn(size, size)
    
    # Warmup
    _ = torch.matmul(a, b)
    
    start = time.time()
    for _ in range(10):
        _ = torch.matmul(a, b)
    end = time.time()
    
    print(f"PyTorch Average Time: {(end - start) / 10.0:.4f} s")

if __name__ == "__main__":
    main()

File: demo_visual/src/bin/benchmark_matmul.rs
use rustorch_core::Tensor;
use std::time::Instant;

fn main() {
    let size = 1024;
    println!("Benchmarking MatMul [{}, {}] ...", size, size);

    // Use ones instead of randn for simplicity if randn is missing
    let a = Tensor::ones(&[size, size]);
    let b = Tensor::ones(&[size, size]);

    // Warmup
    let _ = a.matmul(&b);

    let start = Instant::now();
    for _ in 0..10 {
        let _ = a.matmul(&b);
    }
    let duration = start.elapsed();

    println!("Rust Average Time: {:.4} s", duration.as_secs_f32() / 10.0);
}
